### PR TITLE
Script for restarting testnet cluster.

### PR DESCRIPTION
**Pull request by kriscieslak**

### CODE REVIEW

File: restart-script/restart-cluster [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  genesis_ip=`head -1 $HOSTS_FILE`
+
+  checkNodeState $genesis_ip 9000 "Ready"
+
+  while read ip; do
+    if [ "$ip" == "$genesis_ip" ]; then
+      echo "Skipping genesis node"
+    else
+      echo "Joining node: $ip to $genesis_ip"
+      sleep 15
```
Comment (TheMMaciek) [inline restart-script/restart-cluster:120]: Not sure if you need to wait at all after the recent fixes? 🤔

Comment (kriscieslak) [reply restart-script/restart-cluster:120]: Ok, I think I will move this sleep before `while` loop like in function below, so other nodes have some more time to reach just to reach `ReadyToJoin` state. Unfortunately I cannot check such state like above for genesis node cause `node/state` endpoint is not public.

File: restart-script/cluster-hosts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+13.52.246.74
+52.53.226.167
+54.177.146.221
```
Comment (kpudlik) [inline restart-script/cluster-hosts:3]: Imo it should not be commited

File: restart-script/rollback [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+export CL_EXTERNAL_IP=13.52.246.74 && \
```
Comment (kpudlik) [inline restart-script/rollback:3]: Same here, I would not commit hardcoded IPs

File: restart-script/restart-cluster [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+  while [ "$expected_node_state" != "$actual_node_state" ]; do
+          actual_node_state=$(sshGenesis "curl -s http://$node_to_check:$port_to_check/cluster/info | jq '.[] | select( .ip == \"$node_to_check\" )' | jq '.state' -r")
+    echo -e "${pnk} Current node ($node_to_check) state: $actual_node_state. Expected state: $expected_node_state ${clr}"
+    sleep 1
+  done
+}
+
+stopServices() {
+  echo -e "${blu}Stopping node snd dag services${clr}"
```
Comment (marcinwadon) [inline restart-script/restart-cluster:44]: typo

File: restart-script/restart-cluster [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+  while read -r ip; do
+    echo "Stopping services for: $ip"
+    ssh admin@$ip -o StrictHostKeyChecking=no -n 'sudo systemctl stop dag && sudo systemctl stop node'
+  done <$HOSTS_FILE
```
Comment (marcinwadon) [inline restart-script/restart-cluster:49]: imo it can be simplified with a single pssh command

File: restart-script/restart-cluster [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+}
+
+downloadRollbackSnapshot() {
+  echo -e "${blu}Downloading latest snapshot${clr}"
+
+  sshGenesis 'curl -s https://olubc3p4rc.execute-api.us-west-1.amazonaws.com/global-snapshots/latest | jq '.data.hash' -r > /home/admin/tessellation/rollback-snapshot'
```
Comment (marcinwadon) [inline restart-script/restart-cluster:55]: what's that url? if this is a testnet block explorer, it should be the domain name. I'd assign it to the variable at the beginning of the file btw

**Comment by kpudlik** [comment]
Temporal script for restarting current testnet (based on old dir structure)

**Comment by marcinwadon** [comment]
Deprecated

**Pull request closed without merging**